src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/collector/NodeConfigCache.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.collector;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.Resource;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+/**
+ * we create a thread-safe unbounded cache instance in {@link com.amazon.opendistro.elasticsearch.performanceanalyzer.AppContext}
+ * to store the node config settings from each node. Any RCA vertex in RCA graph can read the node config directly from
+ * this cache instance. The key of this cache is NodeKey + Resource and value is the actual value of the config setting
+ * (i.e. size of write queue capacity)
+ */
+public class NodeConfigCache {
+
+  private static final int LOADING_CACHE_EVICTION_TIMEOUT = 10;
+  private final Cache<NodeConfigKey, Double> nodeConfigCache;
+
+  //unbounded cache with eviction timeout set to 10 mins
+  public NodeConfigCache() {
+    nodeConfigCache =
+        CacheBuilder.newBuilder()
+            .expireAfterWrite(LOADING_CACHE_EVICTION_TIMEOUT, TimeUnit.MINUTES)
+            .build();
+  }
+
+  public void put(NodeKey nodeKey, Resource config, double value) {
+    nodeConfigCache.put(new NodeConfigKey(nodeKey, config), value);
+  }
+
+  public double get(NodeKey nodeKey, Resource config) {
+    Double ret = nodeConfigCache.getIfPresent(new NodeConfigKey(nodeKey, config));
+    if (ret == null) {
+      return Double.NaN;
+    }
```

Comment:
Since this is not going to sql db, I don't think we need `NaN` value and check. Let's instead throw a checked exception here for consumer code safety? (anyway the check has to be handled by the caller; exception will help protect against cases where callers forget the `NaN` check).

Reply:
done, throw an exception if config does not exist in cache